### PR TITLE
Add Koa auto tracing documentation

### PR DESCRIPTION
**Pull request by arthurM2x**
Also working on some auto span on monitoring middleware and routers, might add later

### CODE REVIEW

File: src/collections/_documentation/performance/distributed-tracing.md [hunk 1]
```diff
@@ -614,6 +614,93 @@ app.use(function processItems(req, res, next) {
...
+  const transaction = Sentry.startTransaction({
+    name: `${ctx.method} ${ctx.url}`,
+    op: 'http.server',
+    parentSpanId,
+    traceId,
+  })
+  ctx.__sentry_transaction = transaction
+  await next()
+  
+  // if using koa router, a nicer way to capture transaction using th ematched route
```
Comment (dashed) [inline src/collections/_documentation/performance/distributed-tracing.md:661]:
```suggestion
  // if using koa router, a nicer way to capture transaction using the matched route
```

**Comment by dashed** [comment]
@arthurM2x Nice! Just to confirm, did this setup work for your Koa app?

I tried setting up a simple Koa app using the example you wrote, and added a simple handler:

```js
app.use(async ctx => {
  ctx.body = 'Hello World';
});
```

which generated an error:

```bash
$ node index.js
Error: Express middleware takes 2-4 arguments. Got: 1
    at wrap (/sentry/koa-apm/node_modules/@sentry/apm/dist/integrations/express.js:101:19)
    at /sentry/koa-apm/node_modules/@sentry/apm/dist/integrations/express.js:118:20
```

However, it works fine if I added an un-used 2nd parameter:

```js
app.use(async (ctx, next) => {
  ctx.body = 'Hello World';
});
```

Hence I'm unsure if the `Apm.Integrations.Express`  integration officially supports Koa.

@rhcarvalho  @kamilogorek @HazAT can you confirm?

File: src/collections/_documentation/performance/distributed-tracing.md [hunk 2]
```diff
@@ -614,6 +614,93 @@ app.use(function processItems(req, res, next) {
...
+const Koa = require('koa')
+const app = new Koa()
+
+Sentry.init({
+  dsn: "___PUBLIC_DSN___",
+  integrations: [
+      // enable HTTP calls tracing
+      new Sentry.Integrations.Http({ tracing: true }),
+      // enable Express.js middleware tracing
+      new Apm.Integrations.Express({ app })
```
Comment (dashed) [inline src/collections/_documentation/performance/distributed-tracing.md:633]: `Apm` is undefined here.

**Comment by arthurM2x** [comment]
My bad, copied the code above without removing the express integration part, the middleware works in my koa application

**Comment by dashed** [comment]
@arthurM2x thanks! just a heads up; my colleague, @doralchan,  is in the midst of a major refactor for the Performance and Tracing docs: https://github.com/getsentry/sentry-docs/pull/1782

That might take priority and is slated to be merged soon. The PR should enable any new additions of tracing docs for specific frameworks/libraries such as Koa.

You'll need to rebase your changes after that PR is merged.

**Comment by dashed** [comment]
@arthurM2x Looks like the changes at https://github.com/getsentry/sentry-docs/pull/1782 are merged. would you be able to move the Koa tracing content to this folder? https://github.com/getsentry/sentry-docs/tree/master/src/collections/_documentation/performance-monitoring/configuration

**Comment by arthurM2x** [comment]
Going to close this one as super seeded by https://github.com/getsentry/sentry-docs/pull/1804

**Pull request closed without merging**